src/curve_intersections.rs

```diff
@@ -0,0 +1,1377 @@
+//! Functions used for clipping parametrized curves and finding intersections between them
+//!
+//! Caution: All functions in this file are provisional and may change in the future!
+
+use crate::common::{solve_cubic, solve_linear, solve_quadratic};
+use crate::{
+    Affine, CubicBez, Line, ParamCurve, ParamCurveArclen, ParamCurveDeriv, ParamCurveExtrema,
+    Point, QuadBez, Rect,
+};
+use arrayvec::ArrayVec;
+use std::ops::Range;
+
+/// A trait for descibing a parameterized curve that can be used in the Bezier clipping algorithm
+pub trait ParamCurveBezierClipping:
```

Comment:
It doesn't make sense for this to be pub if the only thing actually exported is a function. I'm also ok with `pub mod curve_intersections`, especially given the disclaimer.

Reply:
I'm not following. As soon as I remove `pub`, I get errors saying that I can't leak a private trait because the functions that use it are public.